tools/observability/resource-utilization-dash.json

```diff
@@ -0,0 +1,632 @@
+{
+  "annotations": {
+    "list": [
+      {
+        "builtIn": 1,
+        "datasource": {
+          "type": "grafana",
+          "uid": "-- Grafana --"
+        },
+        "enable": true,
+        "hide": true,
+        "iconColor": "rgba(0, 211, 255, 1)",
+        "name": "Annotations & Alerts",
+        "type": "dashboard"
+      }
+    ]
+  },
+  "editable": true,
+  "fiscalYearStartMonth": 0,
+  "graphTooltip": 0,
+  "id": 31,
+  "links": [],
+  "panels": [
+    {
+      "datasource": {
+        "type": "prometheus",
+        "uid": "prometheus"
+      },
+      "description": "",
+      "fieldConfig": {
+        "defaults": {
+          "color": {
+            "mode": "thresholds"
+          },
+          "mappings": [],
+          "max": 98,
+          "min": 0,
+          "thresholds": {
+            "mode": "percentage",
+            "steps": [
+              {
+                "color": "green",
+                "value": null
+              },
+              {
+                "color": "purple",
+                "value": 90
+              }
+            ]
+          },
+          "unit": "percent"
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 0,
+        "y": 0
+      },
+      "id": 7,
+      "options": {
+        "minVizHeight": 75,
+        "minVizWidth": 75,
+        "orientation": "auto",
+        "reduceOptions": {
+          "calcs": [
+            "max"
+          ],
+          "fields": "",
+          "values": false
+        },
+        "showThresholdLabels": false,
+        "showThresholdMarkers": true,
+        "sizing": "auto"
+      },
+      "pluginVersion": "10.4.1",
+      "targets": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "prometheus"
+          },
+          "editorMode": "code",
+          "exemplar": true,
+          "expr": "sum(count by (gpu) (DCGM_FI_DEV_GPU_UTIL > 0) or vector(0)) / sum(count by (gpu) (DCGM_FI_DEV_GPU_UTIL)) * 100",
+          "instant": false,
+          "interval": "",
+          "intervalFactor": 1,
+          "legendFormat": "",
+          "refId": "A"
+        }
+      ],
+      "title": "Cluster Allocation",
```

Comment:
It's showing `number of gpus with utilization greater than 0` / `total number of gpus` at "current time"

Grafana has auto refresh off by default - my best guess is the data was stale? I started a trial and notebook, saw cluster utilization at 100%. Then killed both, refreshed the dashboard, and saw cluster utilization at 0%.

Reply:
When we say current time we mean a time range selected? Or do we mean as in an instant? 

I think the chart looks like it using a time range. If the cluster was full but then empty for half the time range I would expect to get 50% not 100% allocated. 

If we do want an instant where we do just care about the current state of the cluster right now I think the query makes sense but it doesn't look like that is what the chart is doing now?

Reply:
When I said "current time" I meant the time range selected in the grafana dashboard.

Obviously a gauge display shows one value instead of a time series. You can choose how the time series is aggregated. This one has "max" ... so we see the max utilization over the time window - I think that might be the confusion. We can update it to "last", which would indicate current utilization or make it more clear and put "max" in the graph title. Do you have a preference?

Reply:
FWIW, my preference is "last" because I think that's most intuitive.

Reply:
If it last can we change the title to something like current cluster gpu allocation?